tests/FilePoolT.cpp

```diff
@@ -0,0 +1,327 @@
+    wait_ms(1100);
+    CHECK(filePoolGlobalObj->getNumLoadedSamples() == 0);
+
+    filePoolGlobalObj.reset();
+    filePoolGlobalObj = sfz::FilePool::getGlobalObject();
+    CHECK(filePoolGlobalObj->getNumLoadedSamples() == 0);
+}
+
+static const int synthCount = 100;
+
+TEST_CASE("[FilePool] Stress Test")
+{
+    struct TestSynthThread {
+        enum Status {
+            Initialized,
+            Waiting1,
+            Loading,
+            Waiting2,
+            Rendering,
+            Posted,
+        };
+
+        TestSynthThread()
+        {
+            CHECK((bool)semBarrier);
+            synth.setSamplesPerBlock(256);
+            thread.reset(new std::thread( &TestSynthThread::job, this ));
+        }
+
+        ~TestSynthThread()
+        {
+            running.store(false, std::memory_order_release);
+            flag.store(false, std::memory_order_release);
+            status.store(Status::Posted, std::memory_order_release);
+            std::error_code ec;
+            semBarrier.post(ec);
+            CHECK(!ec);
+            thread->join();
+        }
+
+        void job()
+        {
+            status.store(Status::Waiting1, std::memory_order_release);
+            semBarrier.wait();
+            CHECK(status.load(std::memory_order_acquire) == Status::Posted);
+            CHECK(flag.load(std::memory_order_acquire) == false);
+            flag.store(true, std::memory_order_release);
+            status.store(Status::Loading, std::memory_order_release);
+            synth.loadSfzFile(fs::current_path() / "tests/TestFiles/looped_regions.sfz");
+            if (running.load(std::memory_order_acquire)) {
+                status.store(Status::Waiting2, std::memory_order_release);
+                while (semBarrier.wait(), running.load(std::memory_order_acquire)) {
+                    CHECK(status.load(std::memory_order_acquire) == Status::Posted);
+                    CHECK(!flag.load(std::memory_order_acquire));
+                    flag.store(true, std::memory_order_release);
+                    status.store(Status::Rendering, std::memory_order_release);
+                    synth.renderBlock(buffer);
+                    status.store(Status::Waiting2, std::memory_order_release);
+                }
+            }
+        }
+
+        void noteOn()
+        {
+            synth.noteOn(0, 60, 100);
+        }
+
+        void noteOff()
+        {
+            synth.noteOff(0, 60, 100);
+        }
+
+        void trigger()
+        {
+            std::error_code ec;
+            semBarrier.post(ec);
+            ASSERT(!ec);
+        }
+
+        sfz::AudioBuffer<float> buffer { 2, 256 };
+        sfz::Synth synth {};
+        std::unique_ptr<std::thread> thread {};
+        RTSemaphore semBarrier { 0 };
+        std::atomic<bool> running { true };
+        std::atomic<Status> status { Status::Initialized };
+        std::atomic<bool> flag { true };
+    };
+
+    std::unique_ptr<TestSynthThread> synthThreads[synthCount];
```

Comment:
Could we make a vector of these?

Reply:
Yes but I prefer using std::unique_ptr personally.
The memory pointer is stable and faster.

Reply:
OK!